pkgs/applications/networking/cluster/kubebuilder/default.nix

```diff
@@ -0,0 +1,62 @@
+{ lib
+, buildGoModule
+, fetchgit
+, installShellFiles
+, makeWrapper
+, go
+, git
+, cacert
+, make ? false }:
+
+buildGoModule rec {
+  pname = "kubebuilder";
+  version = "3.1.0";
+
+  src = fetchgit {
+    url = "https://github.com/kubernetes-sigs/kubebuilder.git";
+    rev = "v${version}";
+    leaveDotGit = true;
+    sha256 = "0g2nicmn889g8z8vlbs571xp2qfjjy1zadycbjzgvdg17kr1swp6";
+  };
+  vendorSha256 = "0zxyd950ksjswja64rfri5v2yaalfg6qmq8215ildgrcavl9974n";
+
+  doCheck = true;
+
+  postPatch = ''
+    substituteInPlace Makefile \
+      --replace "main.kubeBuilderVersion=" "main.kubeBuilderVersion=v${version}"
+  '';
+
+  buildPhase = ''
+    runHook preBuild
+    make build
+    mkdir -p $out/bin
+    cp bin/kubebuilder $out/bin
+    runHook postBuild
+  '';
```

Comment:
Please do not use the upstream makefile but nixpkgs buildGoModule logic and buildFlagsArray instead.

Reply:
Done... but I'm still trying to figure out how to initialize these variables from `buildFlagsArray` (kubebuilder Makefile excerpt):

```
    -X main.gitCommit=$(shell git rev-parse HEAD) \
    -X main.buildDate=$(shell date -u +'%Y-%m-%dT%H:%M:%SZ') \
```

`main.gitCommit` should be doable (and reproducible), I just need to figure out how to evaluate a shell command into a build-time variable.

Perhaps `main.buildDate` just needs to be left out, or set to a fixed time for reproducibility? I can see how it's useful to the upstream developer to know when the binary was built, how does Nix generally approach these kinds of values in builds? Considering taking the timestamp of the git commit used above.

Reply:
Ok, figured it out, `pkgs/applications/virtualization/ignite/default.nix` was helpful.